### ISSUE TITLE
Calling device.launchApp with URL containing an ampersand (&) crashes Detox

### ISSUE THREAD

**Issue by Asvarox** [wix/Detox#2549]
**Describe the bug**
When calling `device.launchApp({ url: 'https://example.com/?query=value&param=value' })`, Detox attempts to run following command:
```
Android/sdk/platform-tools/adb -s emulator-17162 shell am instrument -w -r -e detoxServer ws://localhost:52254 -e detoxSessionId acc32e82-179c-e537-89ca-3fbd1665148a -e detoxURLOverride https://example.com/?query=value&param=value -e debug false com.app.name.test/androidx.test.runner.AndroidJUnitRunner
```
This fails with following message:
```
detox[89271] ERROR: [exec.js/SPAWN_STDERR, #33] /system/bin/sh: -e: not found

detox[89271] ERROR: [exec.js/SPAWN_STDERR, #33]
Error: Argument expected after "https://example.com/?query=value"
```
And any subsequent tests aren't run.

That's because `detoxURLOverride` value isn't escaped in any way. This can be walked-around by escaping the ampersand (or possibly wrapping entire URL in quotes) which breaks the link for iOS and isn't really intuitive. 

**To Reproduce**

- [x] I have tested this issue on the latest Detox release and it still reproduces
```
node_modules/.bin/detox --version
17.14.6
```

Steps to reproduce:
1. Create a test that calls `device.launchApp({ url: 'https://example.com/?query=value&param=value' })` (or any url with `&` in it)
2. Run the test on Android

Repo with bug reproduction (see `e2e/example.spec.js`) https://github.com/Asvarox/DetoxAmpersandRepro

**Expected behavior**
Should attempt to open the app with the link correctly

**Screenshots**
N/A

**Device and Verbose Detox Logs**
[detox-logs.log](https://github.com/wix/Detox/files/5734953/detox-logs.log)

**Environment (please complete the following information):**
 - Detox: 17.14.6
 - React Native: 0.62.2 and 0.63.4
 - Node: v14.15.1
 - Device: Android Emulator API 28
 - OS: MacOS
 - Test-runner (select one): `jest-circus` and `mocha` (tested on both)

**Comment by noomorph** [comment]
Thanks, checking..

**Comment by d4vidi** [comment]
@Asvarox what's the use case that you have where you need to pass in URL parameters? Is it possible that what you're looking for is this - [Detox launch arguments](https://github.com/wix/Detox/blob/master/docs/APIRef.DeviceObjectAPI.md#7-launchargsadditional-process-launch-arguments)?

**Comment by noomorph** [comment]
@d4vidi , anyway, can we try something from the options suggested here? https://stackoverflow.com/questions/13615841/how-do-you-include-ampersands-in-urls-for-adb-shell-am-start

**Comment by d4vidi** [comment]
Perhaps url encoding could work

**Comment by Asvarox** [comment]
Hey, thanks for prompt replies. 

@d4vidi they use is that my app handles links with query params, I don't think launchArgs apply here. 

To give a more specific example, let's say I'm making a YouTube App and I need it to handle links such as these `https://www.youtube.com/watch?t=3&v=somemovieid` (so they open in my app rather than in browser). When writing a test for these links in Detox, I call `device.launchApp({ url: 'https://www.youtube.com/watch?t=3&v=somemovieid' })`. On iOS it's fine, but on Android it breaks.

I ended up escaping the `&` with `\\` when `device.getPlatform()` returns `android` for now. If I provided the escaped link when running iOS tests, it would passthrough `\\&` literally, breaking the tests.

I was playing around with Detox code in my `node_modules/` and wrapping all args values provided to https://github.com/wix/Detox/blob/e25c5db0e451bd3df39def268e9e3571073087ab/detox/src/devices/drivers/android/tools/instrumentationArgs.js#L7 with `'` did the trick as well.

I think it might be worth considering escaping all argument values anyway (I can imagine that links with `$smth` might get expanded etc.)
